livox_ros2_driver/livox_ros2_driver/lddc.cpp

```diff
@@ -718,427 +709,461 @@ void Lddc::onDiagnosticsTimer()
   for (uint32_t i = 0; i < kMaxSourceLidar; ++i) {
     if (lds_->lidars_[i].handle != kMaxSourceLidar) ++lidar_count_;
   }
+
+  if (connected_lidar_count_ != lds_->connected_lidars_.size())
+  {
+    connected_lidar_count_ = lds_->connected_lidars_.size();
+    for (const auto &lidar : lds_->connected_lidars_)
+    {
+      std::string broadcast_code = lidar.first;
+
+      updater_.add(
+          fmt::format("livox_temperature_{}", broadcast_code),
+          std::bind(
+              &Lddc::checkTemperature, this, std::placeholders::_1,
+              broadcast_code));
+
+      updater_.add(
+          fmt::format("livox_internal_voltage_{}", broadcast_code),
+          std::bind(
+              &Lddc::checkVoltage, this, std::placeholders::_1,
+              broadcast_code));
+
+      updater_.add(
+          fmt::format("livox_motor_status_{}", broadcast_code),
+          std::bind(
+              &Lddc::checkMotor, this, std::placeholders::_1,
+              broadcast_code));
+
+      updater_.add(
+          fmt::format("livox_optical_window_{}", broadcast_code),
+          std::bind(
+              &Lddc::checkDirty, this, std::placeholders::_1,
+              broadcast_code));
+
+      updater_.add(
+          fmt::format("livox_firmware_status_{}", broadcast_code),
+          std::bind(
+              &Lddc::checkFirmware, this, std::placeholders::_1,
+              broadcast_code));
+
+      if (check_pps_signal_)
+      {
+        updater_.add(
+            fmt::format("livox_pps_signal_{}", broadcast_code),
+            std::bind(
+                &Lddc::checkPPSSignal, this, std::placeholders::_1,
+                broadcast_code));
+      }
+
+      updater_.add(
+          fmt::format("livox_service_life_{}", broadcast_code),
+          std::bind(
+              &Lddc::checkServiceLife, this, std::placeholders::_1,
+              broadcast_code));
+
+      updater_.add(
+          fmt::format("livox_fan_status_{}", broadcast_code),
+          std::bind(
+              &Lddc::checkFan, this, std::placeholders::_1,
+              broadcast_code));
+
+      updater_.add(
+          fmt::format("livox_time_sync{}", broadcast_code),
+          std::bind(
+              &Lddc::checkTimeSync, this, std::placeholders::_1,
+              broadcast_code));
+
+      updater_.add(
+          fmt::format("livox_connection_{}", broadcast_code),
+          std::bind(
+              &Lddc::checkConnection, this, std::placeholders::_1,
+              broadcast_code));
```

Comment:
How about adding `registerDiagnosticsUpdater()` like this?

```cpp

void Lddc::registerDiagnosticsUpdater(const std::string& broadcast_code) {
    updater_.add(...);
    updater_.add(...);
    updater_.add(...);
    // ...
    registered_code_set_.insert(broadcast_code);
}

void Lddc::onDiagnosticsTimer(){
    // ...
    if (registered_code_set_.count(broadcast_code) == 0) {
        registerDiagnosticsUpdater(broadcast_code);
    }
    // ...
}

```

Reply:
It' better!  I did in https://github.com/tier4/livox_ros2_driver/pull/14/commits/60ddb19a6cb41819bea8a1e6086e0044ecd68960

Reply:
Don't you have to check if it's already registered?

```cpp
    if (registered_code_set_.count(broadcast_code) == 0) {
```

Reply:
Oh sorry, I forgot adding them.

Reply:
https://github.com/tier4/livox_ros2_driver/pull/14/commits/5e305a159edffec6df444ee32f941a0f05af0de5

Reply:
Why `multiset`? I think just `set` is okay.
Also, it's more natural to check this outside `registerDiagnosticsUpdater()` (inside `onDiagnosticsTimer()`).

Reply:
(or `unordered_set` is fine.)

Reply:
No reason actually I did not know just `set`...
OK, it's better. I fixed them.